### PR TITLE
test: magic link

### PR DESCRIPTION
**Pull request by danielgent**
I thnk due to a copy-paste job, the magic link tests were just testing the code :police_officer:

### CODE REVIEW

File: integration-test/flows/magic-link-flow.spec.ts [hunk 1]
```diff
@@ -97,9 +100,9 @@ describe("code-flow", () => {
     expect(idTokenPayload.email).toBe("test@example.com");
     expect(idTokenPayload.aud).toBe("clientId");
 
-    const authCookieHeader = autenticateResponse.headers.get("set-cookie")!;
+    const authCookieHeader = authenticateResponse.headers.get("set-cookie")!;
 
-    // now check silent auth works when logged in with code----------------------------------------
+    // now check silent auth works when logged in with magic link----------------------------------------
```
Comment (danielgent) [inline integration-test/flows/magic-link-flow.spec.ts:105]: Looks like this was all copy-pasted from the code login flow, but not changed to use the magic link

File: integration-test/flows/magic-link-flow.spec.ts [hunk 2]
```diff
@@ -53,36 +53,23 @@ describe("code-flow", () => {
...
-      connection: "email",
-      nonce,
-      response_type,
-      scope,
-      state,
-      redirect_uri,
-      email: "test@example.com",
-      verification_code: otp,
-    });
+    const authenticatePath = link?.split("https://example.com")[1];
```
Comment (danielgent) [inline integration-test/flows/magic-link-flow.spec.ts:58]: this is better. I'm actually using the magic link :smile: 

The test previously was using the code. There was no test of the magic link at all :sunglasses:

File: integration-test/flows/magic-link-flow.spec.ts [hunk 3]
```diff
@@ -53,36 +53,23 @@ describe("code-flow", () => {
...
 
-    // Authenticate using the code
-    const autenticateResponse = await worker.fetch(
-      `/passwordless/verify_redirect?${verifyRedirectQuery.toString()}`,
-      {
-        redirect: "manual",
-      },
-    );
+    // Authenticate using the magic link
+    const authenticateResponse = await worker.fetch(authenticatePath, {
```
Comment (danielgent) [inline integration-test/flows/magic-link-flow.spec.ts:61]: so here we're essentially clicking on the magic link :wink:

File: integration-test/flows/magic-link-flow.spec.ts [hunk 4]
```diff
@@ -53,36 +53,23 @@ describe("code-flow", () => {
...
+      redirect: "manual",
+    });
 
-    if (autenticateResponse.status !== 302) {
+    if (authenticateResponse.status !== 302) {
```
Comment (danielgent) [inline integration-test/flows/magic-link-flow.spec.ts:65]: typo :mag:

File: integration-test/flows/magic-link-flow.spec.ts [hunk 5]
```diff
@@ -130,4 +117,7 @@ describe("code-flow", () => {
       iss: "https://example.com/",
     });
   });
+  // TO TEST
+  // - that can reuse same magic link again
+  // - that a "bad" magic link doesn't work?  Not sure how much time we want to spend on validating access protection... probably a bit for security holes
```
Comment (danielgent) [inline integration-test/flows/magic-link-flow.spec.ts:122]: at some point, we could sync and go through all the comments. See which you think are worth testing

**Review by markusahlstrand — APPROVED** [review]
LGTM

**Pull request merged**